db/upgrade.php

```diff
@@ -182,6 +115,73 @@ function xmldb_tincanlaunch_upgrade($oldversion) {
         upgrade_mod_savepoint(true, 2015033100, 'tincanlaunch');
     }
 
+    if ($oldversion < 2015112702) { //New version in version.php
```

Comment:
It is co-related to the module's version in which there was a major change in the related database tables of the module.
so no matter what version you are in, if you upgrade you will apply only the changes that are relevant to you and the version you are upgrading from.

Not all regular version changes of the module needs to have a corresponding IF statement with a "version date", only those that change the DB.